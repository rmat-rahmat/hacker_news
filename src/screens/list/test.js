/**
 * @format
 */

import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import List from '.';


describe('Test Details screen', () => {

const mockData = [{
  "by": "Tester",
  "descendants": 71,
  "id": 1,
  "kids": [2, 20],
  "score": 111,
  "time": 1175714200,
  "title": "This is test story",
  "type": "story",
  "url": "http://www.getdropbox.com/u/2/screencast.html"
},
{
  "by": "Tester",
  "descendants": 71,
  "id": 1,
  "kids": [2, 20],
  "score": 111,
  "time": 1175714200,
  "title": "This is test story",
  "type": "story",
  "url": "http://www.getdropbox.com/u/2/screencast.html"
},
{
  "by": "Tester",
  "descendants": 71,
  "id": 1,
  "kids": [2, 20],
  "score": 111,
  "time": 1175714200,
  "title": "This is test story",
  "type": "story",
  "url": "http://www.getdropbox.com/u/2/screencast.html"
}
]



it('renders List correctly', () => {
  const tree = renderer.create(
    <List data={mockData} />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
});