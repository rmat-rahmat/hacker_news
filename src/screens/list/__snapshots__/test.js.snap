// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Details screen renders List correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    Array [
      Object {
        "by": "Tester",
        "descendants": 71,
        "id": 1,
        "kids": Array [
          2,
          20,
        ],
        "score": 111,
        "time": 1175714200,
        "title": "This is test story",
        "type": "story",
        "url": "http://www.getdropbox.com/u/2/screencast.html",
      },
      Object {
        "by": "Tester",
        "descendants": 71,
        "id": 1,
        "kids": Array [
          2,
          20,
        ],
        "score": 111,
        "time": 1175714200,
        "title": "This is test story",
        "type": "story",
        "url": "http://www.getdropbox.com/u/2/screencast.html",
      },
      Object {
        "by": "Tester",
        "descendants": 71,
        "id": 1,
        "kids": Array [
          2,
          20,
        ],
        "score": 111,
        "time": 1175714200,
        "title": "This is test story",
        "type": "story",
        "url": "http://www.getdropbox.com/u/2/screencast.html",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  initialNumToRender={20}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "flex": 0.1,
              "fontSize": 15,
            }
          }
        >
          1
        </Text>
        <View
          style={
            Object {
              "flex": 0.9,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 20,
              }
            }
          >
            This is test story
             
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
              (
              getdropbox.com
              )
            </Text>
          </Text>
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 15,
              }
            }
          >
            111
             points || by 
            Tester
             || 
            15 years ago
          </Text>
          <View
            style={
              Object {
                "color": "black",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              accessibilityLabel="view comments"
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#841584",
                      },
                    ]
                  }
                >
                  71 comments
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#aaaaaa",
            "borderBottomWidth": 1,
            "marginHorizontal": 10,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "flex": 0.1,
              "fontSize": 15,
            }
          }
        >
          2
        </Text>
        <View
          style={
            Object {
              "flex": 0.9,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 20,
              }
            }
          >
            This is test story
             
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
              (
              getdropbox.com
              )
            </Text>
          </Text>
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 15,
              }
            }
          >
            111
             points || by 
            Tester
             || 
            15 years ago
          </Text>
          <View
            style={
              Object {
                "color": "black",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              accessibilityLabel="view comments"
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#841584",
                      },
                    ]
                  }
                >
                  71 comments
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#aaaaaa",
            "borderBottomWidth": 1,
            "marginHorizontal": 10,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "flex": 0.1,
              "fontSize": 15,
            }
          }
        >
          3
        </Text>
        <View
          style={
            Object {
              "flex": 0.9,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 20,
              }
            }
          >
            This is test story
             
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
              (
              getdropbox.com
              )
            </Text>
          </Text>
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 15,
              }
            }
          >
            111
             points || by 
            Tester
             || 
            15 years ago
          </Text>
          <View
            style={
              Object {
                "color": "black",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              accessibilityLabel="view comments"
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#841584",
                      },
                    ]
                  }
                >
                  71 comments
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
